### PR TITLE
[WIP] Integrate examples with vega-editor and local docs

### PR DESCRIPTION
**Pull request by willium**
This PR adds a script that indexes all the examples within examples/specs to be used for the test-gallery, and integrates the two.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,10 +18,11 @@
     "postbuild": "uglifyjs vega-lite.js -cm --source-map vega-lite.min.js.map > vega-lite.min.js && npm run schema",
     "build:all": "npm run clean && npm run build && npm test && npm run lint && npm run images",
     "cover": "tsc && istanbul cover node_modules/.bin/_mocha -- --recursive",
-    "clean": "rm -f vega-lite.* vega-lite-schema.json & find src -name '*.js*' -type f -delete & find test -name '*.js*' -type f -delete & find site -name '*.js*' -type f -delete & rm -rf examples/_diff examples/_original examples/_output examples/images",
+    "clean": "rm -f vega-lite.* vega-lite-schema.json & find src -name '*.js*' -type f -delete & find test -name '*.js*' -type f -delete & find site -name '*.js*' -type f -delete & rm -rf examples/_diff examples/_original examples/_output examples/images & rm examples/all-examples.json",
     "deploy": "scripts/deploy.sh",
     "deploy:gh": "scripts/deploy-gh.sh",
     "lint": "tslint -c tslint.json src/*.ts test/*.ts src/**/*.ts test/**/*.ts",
+    "prestart": "node scripts/index-examples",
     "start": "npm run watch & browser-sync start --server --files 'vega-lite.js' --index '_gallery.html'",
```
Comment (domoritz) [inline package.json]: fix the name of the html file here

File: site/static/test-gallery.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+d3.select('#vl-version').text(vl.version);
```
Comment (domoritz) [inline site/static/test-gallery.ts]: next time, make sure you do a move instead of delete and create

File: site/static/test-gallery.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+d3.select('#vg-version').text(vg.version);
+
+// parse url parameters
+var params = function() {
```
Comment (domoritz) [inline site/static/test-gallery.ts]: why do we still have this?

File: site/static/test-gallery.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  return query
+    .split('&')
+    .map(function(x) { return x.split('='); })
+    .reduce(function(a, b) {
+      a[b[0]] = b[1]; return a;
+    }, {});
+};
+
+d3.json('examples/all-examples.json', function(examples) {
+  console.log(examples);
```
Comment (domoritz) [inline site/static/test-gallery.ts]: delete

File: site/static/test-gallery.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      .append('div')
+      .attr('class', 'viz')
+      .attr('id', function(d:string) { return d; });
+
+    vizEnter.append('h3').text(function(d:string) { return d; });
+    vizEnter.append('div').attr('class', 'desc');
+    vizEnter.append('div').attr('class', 'view');
+
+    examples.forEach(function(example) {
+      d3.json('examples/specs/' + example + '.json', function(error, vlSpec) {
```
Comment (domoritz) [inline site/static/test-gallery.ts]: Can you add the name of the file somewhere visible? We often need to know which example is broken and it's hard to find it.

Comment (willium) [inline site/static/test-gallery.ts]: the h3 (the title of each) is also the name of the file.

File: site/static/test-gallery.ts (hunk 2 above)
Comment (willium) [inline site/static/test-gallery.ts]: whoops. will do.

File: site/static/test-gallery.ts (hunk 5 above)
Comment (domoritz) [inline site/static/test-gallery.ts]: Sweet. I guess I should have read your code ;-)

File: scripts/index-examples [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env node
+// Compile a Vega-Lite spec to Vega
+
+'use strict';
+
+var fs = require('fs'),
+  path = require('path');
+
+var dir = __dirname + '/..';
```
Comment (domoritz) [inline scripts/index-examples]: all caps for constants

Comment (willium) [inline scripts/index-examples]: fixed.

**Pull request merged**